latex/EXPRESSIONS

```diff
@@ -0,0 +1,211 @@
+                ]
+              }
+            }
+          }
+        ]
+      , [ "main"
+        , { "type": "assert_non_empty"
+          , "msg": "Entry-point main cannot be empty"
+          , "$1": {"type": "join", "$1": {"type": "var", "name": "main"}}
+          }
+        ]
+      , [ "_"
+        , { "type": "if"
+          , "cond":
+            { "type": "=="
+            , "$1":
+              { "type": "lookup"
+              , "map": {"type": "var", "name": "srcs"}
+              , "key":
+                { "type": "join"
+                , "$1": [{"type": "var", "name": "main"}, ".tex"]
+                }
+              }
+            , "$2": null
+            }
+          , "then":
+            { "type": "fail"
+            , "msg":
+              [ "main not the base name of a .tex file in srcs"
+              , "main has value"
+              , {"type": "var", "name": "main"}
+              , "srcs have file names"
+              , {"type": "keys", "$1": {"type": "var", "name": "srcs"}}
+              ]
+            }
+          }
+        ]
+      , [ "main"
+        , { "type": "if"
+          , "cond":
+            {"type": "==", "$1": {"type": "var", "name": "stage"}, "$2": ""}
+          , "then": {"type": "var", "name": "main"}
+          , "else":
+            { "type": "join"
+            , "separator": "/"
+            , "$1":
+              [ {"type": "var", "name": "stage"}
+              , {"type": "var", "name": "main"}
+              ]
+            }
+          }
+        ]
+      , [ "srcs"
+        , { "type": "to_subdir"
+          , "subdir": {"type": "var", "name": "stage"}
+          , "$1": {"type": "var", "name": "srcs"}
+          }
+        ]
+      , [ "deps"
+        , { "type": "disjoint_map_union"
+          , "msg": "Dependencies may not conflict"
+          , "$1":
+            { "type": "foreach"
+            , "var": "x"
+            , "range": {"type": "var", "name": "deps"}
+            , "body":
+              {"type": "DEP_RUNFILES", "dep": {"type": "var", "name": "x"}}
+            }
+          }
+        ]
+      , [ "tex inputs"
+        , { "type": "to_subdir"
+          , "subdir": "work"
+          , "$1":
+            { "type": "disjoint_map_union"
+            , "msg": "Staging conflict between staged sources and deps"
+            , "$1":
+              [ {"type": "var", "name": "deps"}
+              , {"type": "var", "name": "srcs"}
+              ]
+            }
+          }
+        ]
+      , [ "env"
+        , { "type": "map_union"
+          , "$1":
+            [ { "type": "singleton_map"
+              , "key": "PATH"
+              , "value": "/bin:/usr/bin:/usr/local/bin"
+              }
+            , { "type": "singleton_map"
+              , "key": "SOURCE_DATE_EPOCH"
+              , "value": "0"
+              }
+            , {"type": "var", "name": "env"}
+            ]
+          }
+        ]
+      , ["runner", {"type": "var", "name": "runner"}]
```

Comment:
This is a no-op to set the variable `runner` to the current value of `runner`. Simply drop that clause?

Reply:
right! thanks